core/modules/catalog/helpers/index.ts

```diff
@@ -474,7 +474,8 @@ export function populateProductConfigurationAsync (context, { product, selectedV
         id: selectedOption.value,
         label: selectedOption.label ? selectedOption.label : /* if not set - find by attribute */optionLabel(context.rootState.attribute, { attributeKey: selectedOption.attribute_code, searchBy: 'code', optionId: selectedOption.value })
       }
-      context.state.current_configuration[attribute_code] = confVal
+      const conf = Object.assign({}, context.state.current_configuration, { [attribute_code]: confVal })
+      Vue.set(context.state, 'current_configuration', conf)
```

Comment:
how about just `Vue.set(context.state.current_configuration, attribute_code, confVal)`?

Reply:
Way better! I'll change it.